pooch/core.py

```diff
@@ -27,6 +29,28 @@
 )
 from .downloaders import DOIDownloader, choose_downloader, doi_to_repository
 
+FilePath = t.Union[str, os.PathLike]
+Actions = te.Literal["download", "fetch", "update"]
+
+
+class DownloaderT(te.Protocol):
+    """
+    A class used to define the type definition for the downloader function.
+    """
+
+    # pylint: disable=too-few-public-methods
+    def __call__(  # noqa: E704
+        self,
+        fname: str,
+        action: t.Optional[FilePath],
+        pooch: "Pooch",
+        *,
+        check_only: t.Optional[bool] = None,
+    ) -> t.Any: ...
+
+
+ProcessorT = t.Callable[[str, Actions, "Pooch"], t.Any]
```

Comment:
I think I'd like to separate the public classes from the type variables in different submodules. What about defining a `pooch.typing` submodule where these can live?

On a side note, would it be feasible to name them without the `T`? Or will it create more confusion?

Reply:
I think that would work. The T is my convention but I can remove that as well.